docs/mvc/tutorials/mvc-with-entity-framework/sample/src/ContosoBooks/Controllers/AuthorController.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc;
+using ContosoBooks.Models;
+using Microsoft.Framework.Logging;
+using Microsoft.AspNet.Mvc.Rendering;
+using Microsoft.Data.Entity.Storage;
+using Microsoft.Data.Entity;
+
+// For more information on enabling MVC for empty projects, visit http://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace ContosoBooks.Controllers
+{
+    public class AuthorController : Controller
+    {
+        [FromServices]
+        public BookContext BookContext { get; set; }
+
+        public IActionResult Index()
+        {
+            return View(BookContext.Authors);
+        }
+
+        public async Task<ActionResult> Details(int id)
+        {
+            Author author = await BookContext.Authors
+                .SingleOrDefaultAsync(x => x.AuthorID == id);
```

Comment:
Would be nice to avoid using `x` in every lambda. Recommend using a full name for the parameter, such as `book`, or sometimes that might conflict with a variable in the encapsulating method, so at least abbreviating to `b`.